lib/spack/spack/test/config.py

```diff
@@ -1132,3 +1132,14 @@ def test_single_file_scope_cache_clearing(env_yaml):
     after = scope.get_section('config')
     assert after
     assert before == after
+
+
+def test_internal_config_cache():
+    config = spack.config.Configuration()
+    config.push_scope(spack.config.InternalConfigScope('internal', {
```

Comment:
Can we remove the external `Configuration` wrapper? I think the test would make sense already with just an `InternalConfigScope` that calls its `clear` method, is that correct?

Reply:
I don't know what the public api is for retrieving a value from a config scope object

Reply:
I don't want to write it like the tests above with

```
before = scope.get_section('config')
scope.clear()
after = scope.get_section('config')
assert before == after
```
because it's also true when `id(before) == id(after)` when the `clear()` would mutate the config without rebinding that variable.